src/nft/components/details/detailsV2/RarityGraph.tsx

```diff
@@ -0,0 +1,93 @@
+import Row from 'components/Row'
+import { Trait } from 'nft/types'
+import styled from 'styled-components/macro'
+import { colors } from 'theme/colors'
+
+const RarityBar = styled.div<{ $color?: string }>`
+  background: ${({ $color, theme }) => $color ?? theme.backgroundOutline};
+  width: 2px;
+  height: 10px;
+  border-radius: 2px;
+`
+
+interface RarityValue {
+  threshold: number
+  color: string
+}
+
+enum RarityLevel {
+  VeryCommon = 'Very Common',
+  Common = 'Common',
+  Rare = 'Rare',
+  VeryRare = 'Very Rare',
+  ExtremelyRare = 'Extremely Rare',
+}
+
+const RarityLevels: Map<string, RarityValue> = new Map([
+  [
+    RarityLevel.VeryCommon,
+    {
+      threshold: 0.8,
+      color: colors.gray500,
+    },
+  ],
+  [
+    RarityLevel.Common,
+    {
+      threshold: 0.6,
+      color: colors.green300,
+    },
+  ],
+  [
+    RarityLevel.Rare,
+    {
+      threshold: 0.4,
+      color: colors.blueVibrant,
+    },
+  ],
+  [
+    RarityLevel.VeryRare,
+    {
+      threshold: 0.2,
+      color: colors.purpleVibrant,
+    },
+  ],
+  [
+    RarityLevel.ExtremelyRare,
+    {
+      threshold: 0,
+      color: colors.magentaVibrant,
+    },
+  ],
+])
+
+function getRarityLevel(rarity: number) {
+  switch (true) {
```

Comment:
fun fact: since there are only 5 steps you can actually do something like this which has better performance - at a downside of initial understanding so I don't really recommend doing it

```
function getRarityLevel(rarity: number) {
  switch (Math.floor(rarity * 5)) {
    case (RarityLevels.get(RarityLevel.VeryCommon)?.threshold ?? Number.MAX_SAFE_INTEGER) * 5:
      return RarityLevels.get(RarityLevel.VeryCommon)
    case (RarityLevels.get(RarityLevel.Common)?.threshold ?? Number.MAX_SAFE_INTEGER) * 5:
      return RarityLevels.get(RarityLevel.Common)
    case (RarityLevels.get(RarityLevel.Rare)?.threshold ?? Number.MAX_SAFE_INTEGER) * 5:
      return RarityLevels.get(RarityLevel.Rare)
    case (RarityLevels.get(RarityLevel.VeryRare)?.threshold ?? Number.MAX_SAFE_INTEGER) * 5:
      return RarityLevels.get(RarityLevel.VeryRare)
    case (RarityLevels.get(RarityLevel.ExtremelyRare)?.threshold ?? Number.MAX_SAFE_INTEGER) * 5:
      return RarityLevels.get(RarityLevel.ExtremelyRare)
    default:
      return RarityLevels.get(RarityLevel.VeryCommon)
  }
}
```

Reply:
Does the performance increase come from the comparison of > vs being equally to an int? Interesting to know but I think I lean towards the better readability of the first implementation personally.

Reply:
Yeah it comes from removing the extra branch that is created in the comparison